weaviate/collections/classes/filters.py

```diff
@@ -806,6 +814,88 @@ def not_equal(self, uuid: UUID) -> _FilterValue2:
         )
 
 
+class _FilterByCount(_FilterBase):
+    def __init__(self, target: Optional[_TargetRefs] = None) -> None:
+        self._target = target
+        self._property = None
```

Comment:
wouldn't it make more sense to have target as None and make _property a Union[str, _CountRef]?

Reply:
I agree that this would be clearer so have refactored